bsm_generator/src/bsm_generator_node.cpp

```diff
@@ -0,0 +1,206 @@
+
+    // Reset config
+    config_ = Config();
+
+    // Load parameters
+    get_parameter<double>("bsm_generation_frequency", config_.bsm_generation_frequency);
+    get_parameter<bool>("bsm_id_rotation_enabled", config_.bsm_id_rotation_enabled);
+    get_parameter<int>("bsm_message_id", config_.bsm_message_id);
+    get_parameter<double>("vehicle_length", config_.vehicle_length);
+    get_parameter<double>("vehicle_width", config_.vehicle_width);
+
+    RCLCPP_INFO_STREAM(get_logger(), "Loaded params: " << config_);
+
+    // Register runtime parameter update callback
+    add_on_set_parameters_callback(std::bind(&BSMGenerator::parameter_update_callback, this, std_ph::_1));
+
+    // Setup subscribers
+    pose_sub_ = create_subscription<geometry_msgs::msg::PoseStamped>("pose", 1,
+                                                              std::bind(&BSMGenerator::poseCallback, this, std_ph::_1));
+    accel_sub_ = create_subscription<automotive_platform_msgs::msg::VelocityAccelCov>("velocity_accel_cov", 1,
+                                                              std::bind(&BSMGenerator::accelCallback, this, std_ph::_1));
+    yaw_sub_ = create_subscription<sensor_msgs::msg::Imu>("imu_raw", 1,
+                                                              std::bind(&BSMGenerator::yawCallback, this, std_ph::_1));
+    gear_sub_ = create_subscription<j2735_v2x_msgs::msg::TransmissionState>("transmission_state", 1,
+                                                              std::bind(&BSMGenerator::gearCallback, this, std_ph::_1));
+    speed_sub_ = create_subscription<geometry_msgs::msg::TwistStamped>("ekf_twist", 1,
+                                                              std::bind(&BSMGenerator::speedCallback, this, std_ph::_1));
+    steer_wheel_angle_sub_ = create_subscription<std_msgs::msg::Float64>("steering_wheel_angle", 1,
+                                                              std::bind(&BSMGenerator::steerWheelAngleCallback, this, std_ph::_1));
+    brake_sub_ = create_subscription<std_msgs::msg::Float64>("brake_position", 1,
+                                                              std::bind(&BSMGenerator::brakeCallback, this, std_ph::_1));
+    heading_sub_ = create_subscription<gps_msgs::msg::GPSFix>("gnss_fix_fused", 1,
+                                                              std::bind(&BSMGenerator::headingCallback, this, std_ph::_1));
+    georeference_sub_ = create_subscription<std_msgs::msg::String>("georeference", 1,
+                                                              std::bind(&BSMGenerator::georeferenceCallback, this, std_ph::_1));
+
+    // Setup publishers
+    bsm_pub_ = create_publisher<carma_v2x_msgs::msg::BSM>("bsm_outbound", 5);
+
+    // Set the BSM Message ID 
+    for(size_t i = 0; i < 4; ++i) // As the BSM Message ID is a four-element vector, the loop should iterate four times.
+    {
+      bsm_message_id_.emplace_back(config_.bsm_message_id >> (8 * i));
+    }
```

Comment:
I think this node recently got some updates on develop related to the id that might not have been merged into foxy/develop yet. Please take a look and confirm if this is the most up to date code.

Reply:
Just did another pass on the develop branch and everything in this ported packaged is up to date (including these lines for the `bsm_message_id_`).